File: src/main/java/com/example/usercrud/persistance/repository/UserRepositoryHibernate.java
package com.example.usercrud.persistance.repository;

import com.example.usercrud.business.entity.User;
import com.example.usercrud.business.entity.annotations.Loggable;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.Optional;

@Repository
@Loggable
public interface UserRepositoryHibernate extends JpaRepository<User, Long>, UserRepository {
}

File: src/main/kotlin/cloud/folium/usercrud/persistance/repository/UserRepositoryHibernate.kt
package cloud.folium.usercrud.persistance.repository

import cloud.folium.usercrud.business.entity.User
import cloud.folium.usercrud.business.entity.annotations.Loggable
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface UserRepositoryHibernate : JpaRepository<User, Long>, UserRepository